app/models/rental.rb

```diff
@@ -1,15 +1,23 @@
 class Rental < ApplicationRecord
   belongs_to :user
   belongs_to :product
+  before_save :generate_confirmation_token
+  after_commit :schedule_expiration_notification, on: [:update]
 
   validates :start_date, presence: true, comparison: { less_than_or_equal_to: :end_date }
   validates :end_date, presence: true, comparison: { greater_than_or_equal_to: :start_date }
   validate :product_available
 
+  enum status: { unconfirmed: 1, confirmed: 2, active: 3, archived: 4 }
+
   def total_price
     (end_date - start_date + 1).to_i * product.price
   end
-  
+
+  def schedule_expiration_notification
+    RentalExpirationWorker.perform_async(id) if self.confirmed?
```

Comment:
Daily Job
Check all rentals where Time.now - 5.days < end_date
And send email to selected users